desktop/package/package.gradle

```diff
@@ -7,8 +7,8 @@ task jpackageSanityChecks {
     description 'Interactive sanity checks on the version of the code that will be packaged'
 
     doLast {
-        // Enforce JDK 11 for compiling and building
-        assert JavaVersion.current().isJava11(): "JDK 11 is required"
+        // Enforce JDK 15 for compiling and building
+        assert JavaVersion.current() == JavaVersion.VERSION_15: "JDK 15 is required"
```

Comment:
Do you want to enforce v15, or also allow v15?

If enforce
- this means binaries will be compiled with v15 (not just `jpackage`-d)
- then we should also enforce v15 for development (i.e. not allow anything below, like v10 or v11)

If also allow
- I can look into the right syntax using Gradle toolchains, as cbeams suggested

Reply:
I think to have the choice would be better as long as we don't support any other LTS java version.

Reply:
@cd2357 wrote:
 
> Do you want to enforce v15, or also allow v15?

I'm not sure the distinction between allowing and enforcing is really meaningful in the context of using Gradle's toolchain support.

If we declare that the build should run against a given toolchain, e.g. JDK 15, that's what's going to happen, in every case, unless the developer changes the line of code in the build file making that declaration.

So I'd think we want to just pin the build to JDK 15. That'll reduce the CI build matrix and associated times considerably, as it'll just be about building on JDK 15 across the different major OS / architectures.

Then we just ratchet up the toolchain version every time we're ready to make the jump to JDK 16, 17, etc. And by the way, as per #5835, we're not ready for that move yet!

@ripcurlx wrote:

> we don't support any other LTS java version

What does "support" mean here? I mean we ship a specific JRE version in our packaged binaries, and that's it, right? Would it have any meaning to say we "support" any other JRE than the one we ship with the app? I might be missing something here...

Reply:
It is indeed confusing.

From a birds-eye view, there are three (actually now four) independent "Java versions" we're dealing with:

### 1. The (local) build JDK

  - This is the Java version used by devs to compile, develop and test the codebase locally
  - this can be anything from JDK 11 and up
  - Depends on what devs have installed locally, not explicitly checked or enforced anywhere (but code only compiles with v11 and up)
 
### 2. The (GitHub) build JDK

  - This is the Java version (matrix of versions) used to build on GH
  - It is defined in `build.yml` L15 of this PR to be v11, 11.0.3, v15, v15.0.5 (before this PR, it was only v11)
  - Goal is to see if the code compiles and tests run for these versions

### 3. The (release packaging) build JDK

  - This is the JDK available on the machine that creates the release binaries
  - It is used to compile and build the release Bisq jars
  - It's currently specifically defined to be JDK 11 https://github.com/bisq-network/bisq/blob/b5a43c7f950444f9182affaf853b873d111209b9/desktop/package/package.gradle#L11
(changed by this PR to v15)

### 4. The (release packaging) `jpackager` JDK

  - Defined at https://github.com/bisq-network/bisq/blob/b5a43c7f950444f9182affaf853b873d111209b9/desktop/package/package.gradle#L68
  - It is used exclusively to create the platform-specific release binaries (`jpackager` from earlier JDKs failed for various reasons)
  - This also determines the JRE included in the OS-specific installers and binaries (currently JRE 15)
  - It does not compile or build any jars, instead uses the jars previously compiled in step 3 above

---

As far as I understand, we've been using JDK 11 pretty much everywhere (except for `jpackager` and delivered JRE).

This PR, according to ripcurl's comment above, would extend the versions 1-3 above to support both JDK 11 (LTS) and JDK 15 (non-LTS).

I think this is also what ripcurl means with "as long as we don't support any other LTS java version" (next LTS is JDK 17, which AFAIK we don't support).